### PR TITLE
AC-601: Create phpcs static check for AbstractBlockTest

### PR DESCRIPTION
**Pull request by eliseacornejo**
[AC-601](https://jira.corp.magento.com/browse/AC-601): Create phpcs static check for AbstractBlockTest

This PR is to start moving the legacy static checks we have on Magento2 codebase. This is covering the **AbstractBlock** changes.

### CODE REVIEW

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * See COPYING.txt for license details.
+ */
+namespace Magento2\Sniffs\Legacy;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+class AbstractBlockSniff implements Sniff
+{
+    const CHILD_HTML_METHOD = 'getChildHtml';
```
Comment (sivaschenko) [inline Magento2/Sniffs/Legacy/AbstractBlockSniff.php:13]: Please use `private` access level by default
```suggestion
    private const CHILD_HTML_METHOD = 'getChildHtml';
```

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        $isTheNextMethodGetChildSomething = in_array(
+            $content,
+            [
+                self::CHILD_HTML_METHOD,
+                self::CHILD_CHILD_HTML_METHOD
+            ]
+        );
+        if (!$isTheNextMethodGetChildSomething) {
+            return;
+        }
```
Comment (sivaschenko) [inline Magento2/Sniffs/Legacy/AbstractBlockSniff.php:51]: Please extract this check to `isApplicable` private function
```suggestion
        if (!$this->isApplicable($content)) {
            return;
        }
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @eliseacornejo ! Please see my comments

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php (hunk 1 above)
Comment (eliseacornejo) [reply Magento2/Sniffs/Legacy/AbstractBlockSniff.php]: Done

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php (hunk 2 above)
Comment (eliseacornejo) [reply Magento2/Sniffs/Legacy/AbstractBlockSniff.php]: Done

Comment (sivaschenko) [reply Magento2/Sniffs/Legacy/AbstractBlockSniff.php]: Thanks! Can you please also apply strict types to the class

Comment (eliseacornejo) [reply Magento2/Sniffs/Legacy/AbstractBlockSniff.php]: Also done!

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
+<?php
+declare( strict_types = 1 );
+/**
+ * Copyright © Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento2\Sniffs\Legacy;
```
Comment (sivaschenko) [inline Magento2/Sniffs/Legacy/AbstractBlockSniff.php:7]:
```suggestion
/**
 * Copyright © Magento. All rights reserved.
 * See COPYING.txt for license details.
 */
declare(strict_types=1);

namespace Magento2\Sniffs\Legacy;
```

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        }
+    }
+
+    /**
+     * Return if it is applicable to do the check
+     *
+     * @param String $content
+     * @return bool
+     */
+    private function isApplicable(String $content): bool
```
Comment (sivaschenko) [inline Magento2/Sniffs/Legacy/AbstractBlockSniff.php:72]:
```suggestion
    private function isApplicable(string $content): bool
```

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                $stackPtr,
+                $this->errorCode
+            );
+        }
+    }
+
+    /**
+     * Return if it is applicable to do the check
+     *
+     * @param String $content
```
Comment (sivaschenko) [inline Magento2/Sniffs/Legacy/AbstractBlockSniff.php:69]: Not to mistake with class name
```suggestion
     * @param string $content
```

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php (hunk 3 above)
Comment (eliseacornejo) [reply Magento2/Sniffs/Legacy/AbstractBlockSniff.php]: Done!

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php (hunk 4 above)
Comment (eliseacornejo) [reply Magento2/Sniffs/Legacy/AbstractBlockSniff.php]: Done

File: Magento2/Sniffs/Legacy/AbstractBlockSniff.php (hunk 5 above)
Comment (eliseacornejo) [reply Magento2/Sniffs/Legacy/AbstractBlockSniff.php]: Done

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
@magento import pr to magento-commerce/magento-coding-standard

**Comment by magento-engcom-team** [comment]
@sivaschenko the pull request successfully imported.

**Pull request merged**